src/com/goide/dlv/DlvStackFrame.java

```diff
@@ -148,20 +147,65 @@ public void computePresentation(@NotNull XValueNode node, @NotNull XValuePlace p
           node.setPresentation(icon, presentation, false);
           return;
         }
+
+        if (variable.type.startsWith("struct struct")) {
+          String type = variable.type.replace("struct struct", "struct").trim();
+          String value = variable.value.replace(type,  "").trim();
+          node.setPresentation(icon, type, value, false);
+          return;
+        }
+
+        if (variable.type.startsWith("struct []")) {
+          String type = variable.type.replace("struct []", "[]").trim();
+          String value = variable.value.replaceAll("(.*\\scap:\\s\\d+,\\s)",  "").trim();
+          node.setPresentation(icon, type, value, false);
+          return;
+        }
+
         node.setPresentation(icon, variable.type, variable.value, false);
       }
 
       @Nullable
       private XValuePresentation getPresentation() {
-        String type = variable.type;
+        final String type = variable.type;
         final String value = variable.value;
-        if (NUMBERS.contains(type)) return new XNumericValuePresentation(value);
-        if ("struct string".equals(type)) return new XStringValuePresentation(value);
+        if (NUMBERS.contains(type)) return new XValuePresentation() {
```

Comment:
Such type information adds lots of visual noise. Even in Java we don't have such visible information and we don't have any claim about it for many years.
<img width="497" alt="screenshot 2015-09-22 11 17 53" src="https://cloud.githubusercontent.com/assets/426890/10013633/ba3ec02e-611b-11e5-800c-ee0f6c396f8c.png">